finite-state/shared/src/main/scala/fs2/data/pfsa/PDFA.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019-2022 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.data.pfsa
+
+import Pred.syntax._
+
+import cats.Show
+import cats.syntax.all._
+
+class PDFA[P, T](val init: Int, val finals: Set[Int], val transitions: Map[Int, List[(P, Int)]])(implicit
```

Comment:
Two questions come to my mind here:

1. Should it be `private` to minimise MiMa surface? Are we expecting any user of the library to work with this directly?
2. Are the `Int`s representing the states sequential? Because then we could use an `IndexedSeq` (or even `Array`) for faster access that the `Map`.